scheds/rust/scx_lavd/src/bpf/main.bpf.c

```diff
@@ -312,6 +307,24 @@ static u64 calc_lat_cri(struct task_struct *p, struct task_ctx *taskc,
 	if (is_kernel_task(p))
 		lat_cri += LAVD_LC_KTHREAD_FT;
 
+	/*
+	 * Prioritize a lock holder for faster system-wide forward progress.
+	 */
+	if (is_lock_holder(taskc)) {
+		lat_cri += (lat_cri * LAVD_LC_LOCK_HOLDER_FT) / 1000;
+
+		/*
+		 * Update statistics.
+		 */
+		cpuc_cur->nr_lhp++;
+
+		/*
+		 * Reset task's lock and futex boost count
+		 * for a lock holder to be boosted only once.
+		 */
+		reset_lock_futex_boost(taskc);
```

Comment:
Yes, the counter could become negative so the decrement logic test if the counter is positive before decrementing it.
I think using counter is necessary because a task can hold multiple locks.